src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc

```diff
@@ -363,51 +439,45 @@ class AresClientChannelDNSResolverFactory : public ResolverFactory {
 
 class AresDNSResolver : public DNSResolver {
  public:
+  // Abstract class that centralizes common request handling logic via the
```

Comment:
The interactions between the base class and the subclasses here is fairly complex and hard to follow.  I think this is a symptom of the problem I pointed out in my other comment, where the paradigm of the current c-ares code does not match that of the EE API.  I think if we clean up that underlying problem, then this code will get a lot simpler, because a lot of the state that is currently being stored in the `AresRequest` base class will instead be able to be stored in the `DNSResolver` class itself, which all of the individual request objects will have a ref to.

Reply:
See other thread for agreed-upon terms of a "cleanup" and ownership stories.